hack/update-bazel.sh

```diff
@@ -17,28 +17,10 @@ set -o errexit
 set -o nounset
 set -o pipefail
 
-# https://github.com/kubernetes/test-infra/issues/5699#issuecomment-348350792
-cd $(git rev-parse --show-toplevel)
-
-# Old way of running gazelle and kazel by first go installing them
-deprecated-update() {
-  OUTPUT_GOBIN="./_output/bin"
-  GOBIN="${OUTPUT_GOBIN}" go install ./vendor/github.com/bazelbuild/bazel-gazelle/cmd/gazelle
-  GOBIN="${OUTPUT_GOBIN}" go install ./vendor/k8s.io/repo-infra/kazel
-
-  "${OUTPUT_GOBIN}/gazelle" fix --external=vendored --mode=fix
-  "${OUTPUT_GOBIN}/kazel" --cfg-path=./hack/.kazelcfg.json
-}
-
-# Ensure ./vendor/BUILD.bazel exists
+cd "$(git rev-parse --show-toplevel)"
 mkdir -p ./vendor
-touch "./vendor/BUILD.bazel"
-
-if ! which bazel &> /dev/null; then
-  echo "Bazel is the preferred way to build and test the test-infra repo." >&2
-  echo "Please install bazel at https://bazel.build/ (future commits may require it)" >&2
-  deprecated-update
-  exit 0
-fi
+find vendor -name BUILD -exec rm '{}' +
```

Comment:
I'm guessing `go mod tiny` doesn't prune these?

Reply:
also minor nit: could combine these with `find vendor -name BUILD -o -name BUILD.bazel -exec '{}' +`

Reply:
`go mod tidy` is quite different from `dep prune`. It ensures that `go.mod` and `go.sum` are up-to-date with modules on disk. It mainly adds missing moduels, and remove unused ones.

Reply:
Done